jbmc/src/java_bytecode/java_utils.cpp

```diff
@@ -29,6 +30,15 @@ bool java_is_array_type(const typet &type)
   return is_java_array_tag(to_struct_type(type).get_tag());
 }
 
+bool is_java_string_type(const struct_typet &struct_type)
+{
+  return struct_type.id() == ID_struct &&
```

Comment:
Necessarily true of a `struct_typet`

Reply:
Interesting, I assumed it wasn't because of the
```  PRECONDITION(struct_type.id()==ID_struct);```
in `gen_nondet_struct_init` in the object factory. I removed the equality check that you commented on, and in a new commit I also removed the precondition from the object factory.

Reply:
That is indeed also redundant (I'm guessing it used to take a `typet`)